Bottom: 8565d09f18d8d07fd240b8f94d0f90abe8170d01
Top:    330b3e94dcfc2b4488c43eac9cb107a793adb496
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-02-16 15:51:32 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9760ab0..870908d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1154,7 +1154,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
